mobile/native/src/ln_dlc/node.rs

```diff
@@ -82,151 +85,156 @@ impl Node {
+        // use `lightning-net-tokio`. But we copied this from `p2pderivatives/ldk-sample`
+        if self.inner.dlc_message_handler.has_pending_messages() {
+            self.inner.peer_manager.process_events();
+        }
+
+        Ok(())
+    }
+
+    fn process_incoming_message(&self, node_id: PublicKey, msg: Message) -> Result<()> {
+        match msg {
+            Message::OnChain(_) | Message::Channel(_) => {
+                tracing::debug!(from = %node_id, "Processing DLC-manager message");
+                let resp = self
+                    .inner
+                    .dlc_manager
+                    .on_dlc_message(&msg, node_id)
+                    .map_err(|e| anyhow!(e.to_string()))?;
+
+                if let Some(msg) = resp {
+                    tracing::debug!(to = %node_id, "Sending DLC-manager message");
+                    self.inner.dlc_message_handler.send_message(node_id, msg);
+                }
+            }
+            Message::SubChannel(incoming_msg) => {
+                tracing::debug!(
+                    from = %node_id,
+                    msg = %sub_channel_message_as_str(&incoming_msg),
+                    "Processing DLC channel message"
+                );
+                let reply_msg = self
+                    .inner
+                    .sub_channel_manager
+                    .on_sub_channel_message(&incoming_msg, &node_id)
+                    .map_err(|e| anyhow!(e.to_string()))?;
+
+                if let SubChannelMessage::Offer(offer) = &incoming_msg {
+                    let channel_id = offer.channel_id;
+
+                    // TODO: We should probably verify that: (1) the counterparty is the
+                    // coordinator and (2) the DLC channel offer is expected and correct.
+                    self.inner
+                        .accept_dlc_channel_offer(&channel_id)
+                        .with_context(|| {
+                            format!(
+                                "Failed to accept DLC channel offer for channel {}",
+                                hex::encode(channel_id)
+                            )
+                        })?
+                }
+
+                if let SubChannelMessage::CloseOffer(offer) = &incoming_msg {
+                    let channel_id = offer.channel_id;
+
+                    // TODO: We should probably verify that: (1) the counterparty is the
+                    // coordinator and (2) the DLC channel close offer is expected and correct.
+                    self.inner
+                        .accept_dlc_channel_collaborative_settlement(&channel_id)
+                        .with_context(|| {
+                            format!(
+                                "Failed to accept DLC channel close offer for channel {}",
+                                hex::encode(channel_id)
+                            )
+                        })?;
```

Comment:
Shouldn't this be a match block?

Reply:
For reference: the `if let`s were already there, I just moved them around.

I do agree that a `match` is a bit clearer as one can more easily see that the cases are independent. On the other hand, it does mean that we have to explicitly handle the catch-all pattern explicitly, which is a bit noisy.

Reply:
Improved with https://github.com/get10101/10101/pull/439/commits/412605a2af653877ac1a64510aeae1d8a95c2a4e.